File: 1001-1100/1010-Pairs of Songs With Total Durations Divisible by 60/1010-Pairs of Songs With Total Durations Divisible by 60.cpp
class Solution {
public:
    int numPairsDivisibleBy60(vector<int>& time) {
        vector<int> table(60);
        int total = 0;
        for (int t : time) {
            t %= 60;
            total += table[(60 - t) % 60];
            ++table[t];
        }
        return total;
    }
};

File: 1001-1100/1010-Pairs of Songs With Total Durations Divisible by 60/1010-Pairs of Songs With Total Durations Divisible by 60.py
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        table = [0] * 60
        total = 0
        for t in time:
            t %= 60
            total += table[(60 - t) % 60];
            table[t] += 1
        return total
